src/main/java/com/commercetools/sync/inventories/utils/InventoryReferenceResolutionUtils.java

```diff
@@ -71,24 +75,32 @@ private static InventoryEntryDraft mapToInventoryEntryDraft(@Nonnull final Inven
         return InventoryEntryDraftBuilder
             .of(inventoryEntry)
             .custom(mapToCustomFieldsDraft(inventoryEntry))
-            .supplyChannel(mapToChannelResourceIdentifier(inventoryEntry))
+            .supplyChannel(getResourceIdentifierWithKey(inventoryEntry.getSupplyChannel()))
             .build();
     }
 
-    @Nullable
-    private static ResourceIdentifier<Channel> mapToChannelResourceIdentifier(
-        @Nonnull final InventoryEntry inventoryEntry) {
-        final Reference<Channel> supplyChannelReference = inventoryEntry.getSupplyChannel();
-        if (supplyChannelReference != null) {
-            if (supplyChannelReference.getObj() != null) {
-                return ResourceIdentifier.ofKey(supplyChannelReference.getObj().getKey());
-            }
-            return ResourceIdentifier.ofId(supplyChannelReference.getId());
-        }
-        return null;
+    /**
+     * Builds a {@link InventoryEntryQuery} for fetching inventories from a source CTP project with all the
+     * needed references expanded for the sync:
+     * <ul>
+     *     <li>Custom Type</li>
+     *     <li>Supply Channel</li>
+     * </ul>
+     *
+     * <p>Note: Please only use this util if you desire to sync all the aforementioned references from
+     * a source commercetools project. Otherwise, it is more efficient to build the query without expansions, if they
+     * are not needed, to avoid unnecessarily bigger payloads fetched from the source project.
+     *
+     * @return the query for fetching inventories from the source CTP project with all the aforementioned references
+     *         expanded.
+     */
+    public static InventoryEntryQuery buildInventoryQuery() {
```

Comment:
the idea is giving a inventory all references expanded so sync could do the right resolution for the references, otherwise customer could use the query directly from the SDK if they want to modify for their needs.